src/main/java/io/cdap/plugin/gcp/bigquery/sink/DelegatingMultiSinkOutputCommitter.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.gcp.bigquery.sink;
+
+import io.cdap.cdap.api.data.schema.Schema;
+import org.apache.hadoop.mapreduce.JobContext;
+import org.apache.hadoop.mapreduce.JobStatus;
+import org.apache.hadoop.mapreduce.OutputCommitter;
+import org.apache.hadoop.mapreduce.TaskAttemptContext;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Output Committer which creates and delegates operations to other Bigquery Output Committer instances.
+ *
+ * Delegated instances are supplied along with a schema, which is used to configure the commit operation.
+ */
+public class DelegatingMultiSinkOutputCommitter extends OutputCommitter {
```

Comment:
why are we changing the way the current sink plugin works? I would rather keep the same processing logic for both of them. Why do you need a separate impl here? You will still have an input stream with an Avro record format.

Reply:
By default, the "current" way the plugin works remains the same. The schema gets validated and updated (if the user specifies the Update Schema option) and the records are stored in GCS and then copied to BQ.

The "new" solution will only be used when the users specify the "Allow Flexible Schemas" option. This is used when the record schema is now known at pipeline execution time (Right now, this is only possible when using the Multi Table Source with custom SQL statements: https://github.com/data-integrations/multi-table-plugins/blob/develop/docs/MultiTableDatabase-batchsource.md#custom-sql-statements